pkg/aws/aws.go

```diff
@@ -0,0 +1,144 @@
+package aws
+
+import (
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/autoscaling/autoscalingiface"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type Client struct {
+	// TODO: support multiple ASG
+	ASClient         autoscalingiface.AutoScalingAPI
+	AsgName          string
+	LaunchConfigName string
+}
+
+func (c *Client) AnASGNamed(name string) error {
+
+	sess, err := GetAWSCredentials()
+	if err != nil {
+		return errors.Errorf("Failed getting AWS credentials: %v", err)
+	}
+
+	ASC := autoscaling.New(sess)
+
+	// Simple call to check valid client
+	out, err := ASC.DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
```

Comment:
Does it make more sense to use sts.GetCallerIdentity to validate the client?

Reply:
nvm I think it's just the comment is misleading

Reply:
Ok, I will remove the comment